pkg/controller/elasticsearch/validation/warnings.go

```diff
@@ -26,10 +37,35 @@ func noUnsupportedSettings(es esv1.Elasticsearch) field.ErrorList {
 			errs = append(errs, field.Invalid(field.NewPath("spec").Child("nodeSets").Index(i).Child("config"), es.Spec.NodeSets[i].Config, cfgInvalidMsg))
 			continue
 		}
-		unsupported := config.HasKeys(esv1.UnsupportedSettings)
-		for _, setting := range unsupported {
-			errs = append(errs, field.Forbidden(field.NewPath("spec").Child("nodeSets").Index(i).Child("config").Child(setting), unsupportedConfigErrMsg))
-		}
+		newErrs := check(config, i)
+		errs = append(errs, newErrs...)
+	}
+	return errs
+}
+
+func validateSettings(config *common.CanonicalConfig, index int) field.ErrorList {
+	var errs field.ErrorList
+	unsupported := config.HasKeys(esv1.UnsupportedSettings)
+	for _, setting := range unsupported {
+		errs = append(errs, field.Forbidden(field.NewPath("spec").Child("nodeSets").Index(index).Child("config").Child(setting), unsupportedConfigErrMsg))
+	}
+	return errs
+}
+
+func validateClientAuthentication(config *common.CanonicalConfig, index int) field.ErrorList {
+	type ClientAuthSetting struct {
+		Value string `config:"xpack.security.http.ssl.client_authentication"`
+	}
+	forbiddenValue := "required" // we allow 'none' and 'optional' but 'required' is not supported
+
+	var errs field.ErrorList
+	var setting ClientAuthSetting
+	if err := config.Unpack(&setting); err != nil {
+		return errs
+	}
+	if setting.Value == forbiddenValue {
```

Comment:
Maybe a bit overkill as we only have one setting and one forbidden value but we could create a map in `elasticsearch/v1/fields.go`:

```go
var PartiallyUnsupportedSettings = map[string][]string{
	XPackSecurityHttpSslClientAuthentication: {"required"}, // "optional" or "none" are allowed
}
```

And iterate:

```go
func validPartiallyUnsupportedSettings(config *common.CanonicalConfig) field.ErrorList {
	var errs field.ErrorList
	for setting, unsupportedVals := range esv1.PartiallyUnsupportedSettings {
		value, _ := config.String(setting)
		for _,  := range unsupportedVals {
			if value == unsupportedVal {
				errs = append(errs, field.Invalid(field.NewPath("spec").Child("nodeSets").Index(i).Child("config").Child(setting),
					value, partiallyUnsupportedConfigErrMsg(setting, unsupportedVal)))
			}
		}

	}
	return errs
}
```

Reply:
Nice! I was thinking about the same thing initially 😄. We should probably have something like this in case we encounter another similar setting. But yeah, for one setting only, maybe the simpler is better (for now).